### PR TITLE
Restore Defaults is applied only on OK pressed

### LINKED ISSUE #1415: Shouldn't Preferences > Restore Defaults give the user a chance to confirm?
**Synfig version & platform**:
Building from source
Debian sid

**Issue description**:
Screenshot from 2020-05-16 09-30-36

When I hit the Restore Defaults button, synfig restores everything back to factory settings AND saves it without the user either confirming the changes with OK button or Canceling the changes.

My expected outcome for Restore Defaults is
1. Change to factory default values
2. Wait for the user to either
    - Confirm the change (OK)
                    or
    - Revert the change (Cancel)

### PR DESCRIPTION
**Pull request by FirasH**
Fixes #1415

### CODE REVIEW

**Comment by FirasH** [comment]
Includes also a missing space in the Revert to saved confirm dialog.

**Comment by FirasH** [comment]
There is a functional issue with the current implementation.

Some values have their signal connected to a function which does change the settings of Synfig.
This means that if for example autosave is disabled, when `Cancel` is pressed it won't be reverted to current settings:
https://github.com/synfig/synfig/blob/6d3f75ffc8a3450587c8105a738e1b6db79f7956/synfig-studio/src/gui/dialogs/dialog_setup.cpp#L829-L834

Probably the best way to deal with it is to remove those parts that in practice don't give any advantage to the user.

**Comment by Keyikedalube** [comment]
I think these portions of code are safe to be removed when you do your final code cleanup:
- https://github.com/FirasH/synfig/blob/75fd11d512a23b38910efec2c2dbd35382e9d986/synfig-studio/src/gui/app.cpp#L2271-L2310
- https://github.com/FirasH/synfig/blob/75fd11d512a23b38910efec2c2dbd35382e9d986/synfig-studio/src/gui/app.h#L345

File: synfig-studio/src/gui/instance.cpp [hunk 1]
```diff
@@ -974,7 +974,7 @@ Instance::safe_revert()
 	{
 		if (!App::dialog_message_2b(
 			_("Revert to saved"),
-			_("You will lose any changes you have made since your last save."
+			_("You will lose any changes you have made since your last save. "
```
Comment (ice0) [inline synfig-studio/src/gui/instance.cpp:977]: Is extra space needed?

Comment (rodolforg) [reply synfig-studio/src/gui/instance.cpp:977]: It's because "are you share" ;)
Maybe a newline `\n` would be better.

Comment (FirasH) [reply synfig-studio/src/gui/instance.cpp:977]: Without the space here is how they look:
no_space

Comment (ice0) [reply synfig-studio/src/gui/instance.cpp:977]: New line as suggested by @rodolforg would be better, but that is ok too.

**Comment by FirasH** [comment]
@ice0 this request is ready. Do you need me to force-push it?

**Comment by ice0** [comment]
@FirasH yes, can you rebase it?

**Comment by FirasH** [comment]
> @FirasH yes, can you rebase it?

Done :+1:

**Comment by ice0** [comment]
Merged. Thank you!

**Pull request merged**